pkg/dnsutil/hostsstore/hostsstore.go

```diff
@@ -114,6 +114,10 @@ func (x *hostsStore) Acquire(meta Meta) (err error) {
 		if err = os.WriteFile(loc, []byte{}, 0o644); err != nil {
 			return errors.Join(store.ErrSystemFailure, err)
 		}
+		// See https://www.man7.org/linux/man-pages/man2/open.2.html
```

Comment:
This doesn't explain how the umask rule differs across Go's `os.WriteFile` and `os.Chmod`

Reply:
No chmod implementation (system or language) would ever honor change the provided mode to account for umask - that would not make sense.

As for WriteFile, it calls `syscall.Open` underneath - and Open behavior wrt umask is covered in the man page.

Anyhow, changed the comment.